src/System.Net.Http/src/System/Net/Http/Managed/ManagedHandler.cs

```diff
@@ -309,11 +309,19 @@ private HttpMessageHandler SetupHandlerChain()
                 request.Headers.TransferEncodingChunked = true;
             }
 
-            if (request.Version.Minor == 0 && request.Version.Major == 1 &&
-                request.HasHeaders && request.Headers.TransferEncodingChunked == true)
+            if (request.Version.Minor == 0 && request.Version.Major == 1 && request.HasHeaders)
             {
                 // HTTP 1.0 does not support chunking
-                return Task.FromException<HttpResponseMessage>(new NotSupportedException(SR.net_http_unsupported_chunking));
+                if (request.Headers.TransferEncodingChunked == true)
+                {
+                    return Task.FromException<HttpResponseMessage>(new NotSupportedException(SR.net_http_unsupported_chunking));
+                }
+
+                // HTTP 1.0 does not support Expect: 100-continue; just disable it.
+                if (request.Headers.ExpectContinue == true)
+                {
+                    request.Headers.ExpectContinue = false;
```

Comment:
Modifying this property is another case of the handler modifying the request message itself.  Perhaps in this case, it's ok.  But we should document this since for the most part, the HttpRequestMessage passed to the handler should be immutable except for specifically documented things, i.e. RequestUri property modified in cases of redirect.

See: #25163